whichone/static/js/main.js

```diff
@@ -87,3 +89,10 @@ function initOptions() {
   valenceBox.checked = true;
   durationBox.checked = true;
 }
+
+
+function spotifyLogout() {
+  const url = 'https://accounts.spotify.com/en/logout';
+  const spotifyLogoutWindow = window.open(url, 'Spotify Logout', 'width=700,height=500,top=40,left=40');
+  setTimeout(() => spotifyLogoutWindow.close(), 2000);
+}
```

Comment:
Have you checked if Spotify supports CORS? If yes you could do an AJAX request and avoid popping up a window.